### ISSUE TITLE
 Generalization problem

### ISSUE THREAD

**Issue by qgh1223** [nibtehaz/MultiResUNet#3]
Have you tried some datasets with separate valid dataset and test dataset?I found that I got a good result in train dataset and a bad result in valid dataset in ISIC 2016 and ISIC 2017

**Comment by nibtehaz** [comment]
Unfortunately, No. I haven't done so. I used the dataset from ISIC 2018 competition, but at the time of my work the validation and test labels were not published.

Since the algorithm was evaluated on all the datasets using a 5-fold cross validation scheme, I don't think it is that much prone to overfitting. Perhaps, the validation and the test sets of the ISIC 2016 and ISIC 2017 are drastically different from the training set. Also, since this is a competition some innovative pre-processing, post-processing or even different learning schemes and parameter tweaking should be done for attaining an overall better result. Maybe the imaging devices and/or image acquisition methods were different as well, and some domain specific pre-processing like color constancy (http://vislab.isr.ist.utl.pt/wp-content/uploads/2012/12/14-ICIPa.pdf) would make the images more similar.
